sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -539,6 +540,10 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
       index > beginIndex && !s"${line.charAt(index)}".trim.isEmpty)
 
     for (index <- 0 until line.length) {
+      if (index > 0 && index - 1 == bracketedCommentRightBound) {
```

Comment:
Please leave some comments here about why we need to decrement `bracketedCommentLevel` at the beginning of this loop?